### PR TITLE
Add metrics backend for SignalFx

### PR DESCRIPTION
**Pull request by nbroyles**
Wire up SignalFx for a metric reporting option. Additionally, pull in the jackson-jsr310 dependency so that we can properly parse yaml configs for duration. Without this, loading duration from yaml will crash.

### CODE REVIEW

File: misk/src/main/kotlin/misk/metrics/backends/signalfx/SignalFxBackendModule.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        binder().addMultibinderBinding<Service>().to<SignalFxReporterService>()
+    }
+
+    @Provides
+    @Singleton
+    fun signalFxReporter(
+            @AppName appName: String,
+            config: SignalFxBackendConfig,
+            metricRegistry: MetricRegistry
+            ): SignalFxReporter {
```
Comment (mmihic) [inline misk/src/main/kotlin/misk/metrics/backends/signalfx/SignalFxBackendModule.kt]: nit: this indentation seems weird - are you using the standard codestyle from square/codestyles?

File: misk/src/test/kotlin/misk/config/ConfigTest.kt [hunk 2]
```diff
@@ -33,6 +34,9 @@ class ConfigTest {
     @field:[Inject Named("consumer_b")]
     lateinit var consumerB: ConsumerConfig
 
+    @field:[Inject]
+    lateinit var durationConfig: DurationConfig
```
Comment (ryanhall07) [inline misk/src/test/kotlin/misk/config/ConfigTest.kt]: why inject this?

File: misk/src/main/kotlin/misk/metrics/backends/signalfx/SignalFxBackendModule.kt (hunk 1 above)
Comment (nbroyles) [reply misk/src/main/kotlin/misk/metrics/backends/signalfx/SignalFxBackendModule.kt]: I am. Not sure what happened here, but cleaned it up.

File: misk/src/test/kotlin/misk/config/ConfigTest.kt (hunk 2 above)
Comment (nbroyles) [reply misk/src/test/kotlin/misk/config/ConfigTest.kt]: Good catch. Don't need it.

**Pull request merged**